From f71c72b11068bd5ef3a97e85756ff337aa313935 Mon Sep 17 00:00:00 2001
From: hscham <hscham@chromium.org>
Date: Mon, 14 Dec 2020 10:48:12 +0900
Subject: [PATCH] libchrome: remove deprecated version of
 base::Value::RemovePath

Would be deprecated after r822064 uprev.
Use base::Value::RemovePath(StringPiece path) instead.

Change-Id: I9df48392cc313f8928dee8de9a0a1322832fb4a3
---
 base/values.cc | 23 -----------------------
 base/values.h  |  4 ----
 2 files changed, 27 deletions(-)

diff --git a/base/values.cc b/base/values.cc
index 5f5657935d..08d34131bd 100644
--- a/base/values.cc
+++ b/base/values.cc
@@ -797,29 +797,6 @@ Value* Value::SetPath(span<const StringPiece> path, Value&& value) {
   return cur->SetKey(*cur_path, std::move(value));
 }
 
-bool Value::RemovePath(std::initializer_list<StringPiece> path) {
-  DCHECK_GE(path.size(), 2u) << "Use RemoveKey() for a path of length 1.";
-  return RemovePath(make_span(path.begin(), path.size()));
-}
-
-bool Value::RemovePath(span<const StringPiece> path) {
-  if (!is_dict() || path.empty())
-    return false;
-
-  if (path.size() == 1)
-    return RemoveKey(path[0]);
-
-  auto found = dict_.find(path[0]);
-  if (found == dict_.end() || !found->second->is_dict())
-    return false;
-
-  bool removed = found->second->RemovePath(path.subspan(1));
-  if (removed && found->second->dict_.empty())
-    dict_.erase(found);
-
-  return removed;
-}
-
 Value::dict_iterator_proxy Value::DictItems() {
   CHECK(is_dict());
   return dict_iterator_proxy(&dict_);
diff --git a/base/values.h b/base/values.h
index ffd2c590e4..cc68b226f9 100644
--- a/base/values.h
+++ b/base/values.h
@@ -435,10 +435,6 @@ class BASE_EXPORT Value {
   //   bool success = value.RemovePath("foo.bar");
   bool RemovePath(StringPiece path);
 
-  // Deprecated versions
-  bool RemovePath(std::initializer_list<StringPiece> path);
-  bool RemovePath(span<const StringPiece> path);
-
   // Tries to extract a Value at the given path.
   //
   // If the current value is not a dictionary or any path component does not
-- 
2.29.2.684.gfbc64c5ab5-goog

